File: in_typescript/src/day_01_part_2.ts
import { readFileSync } from 'fs';

let input: number[] = readFileSync('../inputs/day01.txt', "utf8").
    split("\n").
    map(i => parseInt(i)).
    slice(0, -1);

let massToFuel = function(mass: number): number {
    let fuel = Math.floor(mass / 3) - 2
    return fuel > 0 ? fuel + massToFuel(fuel) : 0
}

let fuel = input.map(massToFuel).reduce((a, i) => a + i)

console.log("Day 01 - Part 2:", fuel)

File: in_clojure/day_01_part_2.clj
(require '[clojure.string :as str])

(defn mass->fuel [mass]
  (let [fuel-mass (- (quot mass 3) 2)]
    (if (> fuel-mass 0)
      (+ fuel-mass (mass->fuel fuel-mass))
      0)))

(->> "../inputs/day01.txt"
     (slurp)
     (str/split-lines)
     (map read-string)
     (map mass->fuel)
     (reduce +)
     (str "Day 01 - Part 2: ")
     (println))
